src/server/game/Entities/Creature/SummonInfo/SummonInfoArgs.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _SummonInfoArgs_h__
+#define _SummonInfoArgs_h__
+
+#include "Define.h"
+#include "Duration.h"
+#include "Optional.h"
+#include "ObjectGuid.h"
+
+struct SummonPropertiesEntry;
+
+struct SummonInfoArgs
+{
+    Optional<ObjectGuid> SummonerGUID;
```

Comment:
default initialization to std::nullopt of all members of this struct maybe?

Reply:
Optional is nullopt by default

Reply:
That is if it is default-constructed, however, by explicitly initializing it with std::nullopt it's more clear to other people that might work on the API on later stages.

Reply:
IMO that's really just nitpicking as anyone who works with that struct should have at least a basic understanding of how the types work that he wants to touch.